recipes/thrift/all/conanfile.py

```diff
@@ -0,0 +1,141 @@
+import os
+from conans import tools, CMake, ConanFile
+
+
+class ConanFileDefault(ConanFile):
+    name = "thrift"
+    description = "Thrift is an associated code generation mechanism for RPC"
+    topics = ("conan", "thrift", "serialization", "rpc")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apache/thrift"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*.diff"]
+    generators = "cmake", "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_libevent": [True, False],
+        "with_openssl": [True, False],
+        "with_boost_functional": [True, False],
+        "with_boost_smart_ptr": [True, False],
+        "with_boost_static": [True, False],
+        "with_boostthreads": [True, False],
+        "with_stdthreads": [True, False],
+        "with_c_glib": [True, False],
+        "with_cpp": [True, False],
+        "with_java": [True, False],
+        "with_python": [True, False],
+        "with_haskell": [True, False],
+        "with_plugin": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "with_libevent": True,
+        "with_openssl": True,
+        "with_boost_functional": False,
+        "with_boost_smart_ptr": False,
+        "with_boost_static": False,
+        "with_boostthreads": False,
+        "with_stdthreads": True,
+        "with_c_glib": False,
+        "with_cpp": True,
+        "with_java": False,
+        "with_python": False,
+        "with_haskell": False,
+        "with_plugin": False
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "thrift-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for p in self.conan_data["patches"][self.version]:
+            tools.patch(**p)
+        for f in ["Findflex.cmake", "Findbison.cmake"]:
+            if os.path.isfile(f):
+                os.unlink(f)
```

Comment:
This surprise me a lot. If we remove these files, why do we want the dependencies? If it is something that is being added via `conan_basic_setup()`, what is it? We are trying to decouple dependencies information from the toolchain, so in the future this might lead to some error.

Reply:
because with these files, this error happens: https://c3i.jfrog.io/c3i/misc/logs/pr/1700/7/thrift/0.13.0/345d69bba8340540d4afe092084e34d4cc2df23f/create_stdout.txt 
We need flex and bison, but files generated by cmake_find_package generator are not conform to upstream, so cannot rely on them.

Reply:
Where does this package find `flex` and `bison`? I see [here](https://github.com/apache/thrift/blob/v0.13.0/compiler/cpp/CMakeLists.txt#L25) it is looking for them.

Reply:
it is using the cmake files provided by cmake https://cmake.org/cmake/help/latest/module/FindFLEX.html and https://cmake.org/cmake/help/latest/module/FindBISON.html

Reply:
🤦 Thanks!